runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingPipelineTranslator.java

```diff
@@ -211,6 +254,23 @@ boolean canTranslate(T transform, FlinkStreamingTranslationContext context) {
         if (WriteFilesTranslation.isWindowedWrites(transform)) {
           replacement = replacement.withWindowedWrites();
         }
+
+        if (options.isAutoBalanceWriteFilesShardingEnabled()) {
+          Preconditions.checkArgument(
+              options.getParallelism() > 0,
+              "Parallelism is required to be set in FlinkPipelineOptions when isAutoBalanceWriteFilesShardingEnabled");
+          Preconditions.checkArgument(
+              options.getMaxParallelism() > 0,
+              "MaxParallelism is required to be set in FlinkPipelineOptions when isAutoBalanceWriteFilesShardingEnabled");
```

Comment:
Why is this required, couldn't we assume `maxParallelism == parallelism` if not provided?

Reply:
I think Flik does use `KeyGroupRangeAssignment .computeDefaultMaxParallelism()` and not directly parallelism. I guess I was just being conservative. I will make it work with defaults

Reply:
Ah that's true. I wasn't aware that we set a higher max parallelism, just in case the user wants to scale up.